cli/src/main/java/com/devonfw/tools/ide/merge/xmlmerger/XmlMergeStrategy.java

```diff
@@ -0,0 +1,149 @@
+    }
+  },
+
+  /**
+   * Keeps the existing target element intact if the source element exists in the target document, otherwise, it is appended.
+   */
+  KEEP {
+    @Override
+    protected void doMerge(Element templateElement, Element resultElement, ElementMatcher matcher) {
+
+      // Do nothing, keep the existing element
+    }
+  };
+
+  /**
+   * @param templateElement the {@link Element} of the template XML file to merge.
+   * @param resultElement the {@link Element} populated with the workspace XML file to merge into.
+   * @param matcher the {@link ElementMatcher}.
+   */
+  public void merge(Element templateElement, Element resultElement, ElementMatcher matcher) {
+
+    try {
+      doMerge(templateElement, resultElement, matcher);
+    } catch (RuntimeException e) {
+      throw new IllegalStateException("Merge strategy " + this + " failed on " + XmlMergeSupport.getXPath(templateElement), e);
+    }
+  }
+
+  protected abstract void doMerge(Element templateElement, Element resultElement, ElementMatcher matcher);
+
+  /**
+   * Returns the MergeStrategy enum constant with the specified name.
+   *
+   * @param name the name of the enum constant to return
+   * @return the enum constant with the specified name
+   */
+  public static XmlMergeStrategy of(String name) {
+
+    return Enum.valueOf(XmlMergeStrategy.class, name.toUpperCase(Locale.ROOT));
+  }
+
+  /**
+   * Combines child nodes (elements, text and CDATA) from the template into the result {@link Element}.
+   *
+   * @param templateElement the template {@link Element}.
+   * @param resultElement the result {@link Element}.
+   * @param elementMatcher the {@link ElementMatcher} used for matching elements.
+   */
+  protected void combineChildNodes(Element templateElement, Element resultElement, ElementMatcher elementMatcher) {
+
+    NodeList templateChildNodes = templateElement.getChildNodes();
+    for (int i = 0; i < templateChildNodes.getLength(); i++) {
+      Node templateChild = templateChildNodes.item(i);
+      if (templateChild.getNodeType() == Node.ELEMENT_NODE) {
+        Element templateChildElement = (Element) templateChild;
+        Element matchedResultElement = elementMatcher.matchElement(templateChildElement, resultElement);
+        if (matchedResultElement != null) {
+          XmlMergeStrategy mergeStrategy = XmlMergeSupport.getMergeStrategy(templateChildElement);
+          if (mergeStrategy == null) {
+            mergeStrategy = this; // fallback "this" will always be COMBINE
+          }
+          mergeStrategy.merge(templateChildElement, matchedResultElement, elementMatcher);
+        } else {
+          Node resultChildElement = resultElement.getOwnerDocument().importNode(templateChildElement, true);
+          resultElement.appendChild(resultChildElement);
+        }
+      } else if (XmlMergeSupport.isTextual(templateChild)) {
+        if (!templateChild.getTextContent().isBlank()) {
+          // TODO this looks wrong to me...
```

Comment:
Needs to be discussed. Either we have to refactor this or we can clarify and just remove the comment:
```suggestion
```

Reply:
But this is not what is implemented here.
*Every* non-empty text node will be replaced using this method and replace the text into the first non-blank text node.
So if you have
template: Text(text1) CDTATA(text2) Text(text3)
workspace: Text(text4) CDATA(text5) Text(text6)
you will create
1. iteration: Text(text1) CDTATA(text5) Text(text6)
2. iteration: Text(text2) CDTATA(text5) Text(text6)
3. iteration: Text(text3) CDTATA(text5) Text(text6)

Lets replace the TODO with a new bug issue.

Reply:
Created #943